### ISSUE TITLE
Why apply .editorconfig every time window is activated?

### ISSUE THREAD

**Issue by mcnameej** [editorconfig/editorconfig-vscode#231]
Please fill-in this template.

- [X] I have a question that is specific to this extension; thus, inappropriate for the main [EditorConfig issue tracker](https://github.com/editorconfig/editorconfig/issues).
- [ ] I tried running `code --disable-extensions` and the issue did **NOT** present itself.

## Issue

|             | Visual Studio Code | editorconfig-vscode |
|-------------|--------------------|---------------------|
| **Version** | `1.34.0`            | `0.13.0`             |

### Question

Why does the VS Code editorconfig plugin re-apply the configuration every time a window is activated?  I understand this was a deliberate choice, and I'm trying to understand the rationale.

This design makes it hard to review third-party code that doesn't match your global configuration.  In particular, you can change the tab size in VS Code so the file displays correctly, but the change is undone as soon as you switch windows.

**Comment by jedmao** [comment]
Without looking at the code, I recall it was done this way because the vscode APIs made available for changing indentation and such were applied not just to a particular code window (tab), but to the entire vscode instance. As such, I implemented it such that when you focus into a particular tab, the EC settings are applied. When you blur out, they are restored to whatever they were before and when you tab into another file, that file's EC settings are loaded.

Furthermore, there have been issues where one edits the `.editorconfig` file with some tabs already open. I wanted to make sure that when they focus back into a particular file that the _new_ EC settings are applied accurately instead of the stale settings being applied (before the `.editorconfig` file was changed).

Does that make sense?

Basically, there were a whole host of issues that would happen if you didn't reload the EC config on every focus. If the vscode APIs allowed me to apply on a per-file basis, I would still have to do this to an extent, because I don't know if you've changed your `.editorcofig` file(s) between the time you blurred out and focused back into a particular tab.

**Comment by jedmao** [comment]
Since this was a question and I answered it, I'm closing for now. We can reopen if there's something more to add.

**Comment by mcnameej** [comment]
> Does that make sense?

Yes.  Thank you for the explanation!

**Comment by jedmao** [comment]
Refer to https://github.com/editorconfig/editorconfig-vscode/wiki/How-it-Works
